libsolidity/formal/SMTEncoder.cpp

```diff
@@ -357,25 +357,17 @@ void SMTEncoder::endVisit(Assignment const& _assignment)
 		Token::AssignSub,
 		Token::AssignMul,
 		Token::AssignDiv,
-		Token::AssignMod
+		Token::AssignMod,
+		Token::AssignBitAnd,
+		Token::AssignBitOr,
+		Token::AssignBitXor,
+		Token::AssignShl,
+		Token::AssignShr,
+		Token::AssignSar
 	};
 	Token op = _assignment.assignmentOperator();
 	if (op != Token::Assign && !compoundOps.count(op))
-	{
-		Expression const* identifier = &_assignment.leftHandSide();
-		if (auto const* indexAccess = dynamic_cast<IndexAccess const*>(identifier))
-			identifier = leftmostBase(*indexAccess);
-		// Give it a new index anyway to keep the SSA scheme sound.
-		solAssert(identifier, "");
-		if (auto varDecl = identifierToVariable(*identifier))
-			m_context.newValue(*varDecl);
-
-		m_errorReporter.warning(
-			9149_error,
-			_assignment.location(),
-			"Assertion checker does not yet implement this assignment operator."
-		);
-	}
+		solAssert(false, "");
```

Comment:
Now that the `set` above is completed with all the assignment operators, we could replace the section by
```
solAssert(TokenTraits::isAssignmentOp(op), "");
```